logic.lua

```diff
@@ -1,167 +1,406 @@
---local MAX_GROUPS = 8
+local Logic = {}
 
-BuffDuty.max_group = 1
+-- Upvalues
+local utils = BuffDuty.Utils
 
-function BuffDuty:getNameClassGroup(idx)
-    local name, r, sg, lvl, cls_loc, cls = GetRaidRosterInfo(idx)
-    if sg > BuffDuty.max_group then
-        BuffDuty.max_group = sg
+-- Generates and returns a duty list in the following format:
+-- list[name]["name"] = the players name
+-- list[name]["i"] = the players index truncated to between 1 and 8 (inclusive)
+-- list[name]["groups"] = a string list of assigned groups
+-- list[name]["s"] = plural modifier; "s" if the player is assigned more than 1 group
+function BuffDuty.generateDuties(cmd, raid_info, class_players)
+    -- No players to assign :(
+    if (class_players.count == 0) then
+        return {} 
+    end
+    -- One player to assign :/
+    if (class_players.count == 1) then
+        local name = next(class_players.map)
+        local duty = {}
+        duty["name"] = name
+        duty["i"] = 1
+        if raid_info.group_count == 1 then
+            duty["s"] = ""
+            duty["groups"] = string.format("%d", raid_info.group_min)
+        else
+            duty["s"] = "s"
+            duty["groups"] = string.format("%d - %d", raid_info.group_min, raid_info.group_max)
+        end
+        local duty_list = {}
+        duty_list[name] = duty
+        return duty_list
+    end
+
+    local player_duty_map = nil
+    if cmd.assign then
```

Comment:
what are the options? Except assign?

Reply:
Default? ... I was going to add more, but really Assign and Default - with the option to disable own group assignment - is pretty much it; unless you have some different assignment logic you want to use?